### PR TITLE
controller: Delete data volumes when one-off jobs stop 

### PR DESCRIPTION
**Pull request by lmars**
I think eventually we want volume GC at a higher level than flynn-host, but for now this prevents leaking data volumes which we know are ephemeral.

### CODE REVIEW

**Comment by titanous** [comment]
Should this have a test?

**Comment by lmars** [comment]
@titanous test added

File: test/test_host.go [hunk 1]
```diff
@@ -471,6 +473,41 @@ func (s *HostSuite) TestNotifyOOM(t *c.C) {
...
+		t.Assert(err, c.IsNil)
+
+		// run the job
+		cmd := exec.JobUsingCluster(s.clusterClient(t), exec.DockerImage(imageURIs["test-apps"]), job)
+		out, err := cmd.CombinedOutput()
+		t.Assert(err, c.IsNil)
+		t.Assert(string(out), c.Equals, "/foo\n")
+
+		// check if the volume was deleted or not
+		vol, err = h.GetVolume(vol.ID)
```
Comment (titanous) [inline test/test_host.go]: This may need to be wrapped in retries as I believe the cleanup is async.

**Comment by lmars** [comment]
@titanous I added a "cleanup" job event

**Pull request merged**